example/istio/README.md

```diff
@@ -0,0 +1,133 @@
+# Coraza Proxy WASM as WasmPlugin for Istio
+
+WasmPlugins allow the Istio proxy to be enhanced with WebAssembly filters. 
+The coraza proxy wasm acts as one of these filters, adding WAF features to Istio. 
+The execution order within Envoy's filter chain is set by phase and priority, facilitating 
+intricate interactions between user-provided WasmPlugins and Istio's built-in filters.
+
+## Istio Setup
+
+Given a multitude of possible Istio setups, we will only cover the most common one with the following assumptions:
+
+- Istio is installed in the `istio-system` namespace
+- The mesh has a singular entrypoint served by the `istio-ingressgateway` service
+- The mesh has an egress gateway served by the `istio-egressgateway` service
+- Traffic is routed to kubernetes namespaces via Istio's `VirtualService` and `DestinationRule` resources
+- Services served by a `VirtualService` have an `istio-proxy` sidecar
+
+## Getting started
+
+The coraza proxy wasm can filter traffic inside the mesh at multiple locations.
+
+### At Ingress gateway for all incoming traffic
+
+The envoy pod of the ingress-gateway can be configured to use the coraza proxy wasm as a filter, thus 
+filtering all incoming traffic.
+
+The following example shows how to configure embedded [Rule Sets](https://github.com/coreruleset/coreruleset)
+at the ingress gateway and use the coraza proxy wasm as a filter.
+
+It utilizes the 
+[WasmPlugin](https://istio.io/latest/docs/reference/config/proxy_extensions/wasm-plugin/) resource of Istio.
+This way the filter can be configured via the `pluginConfig` field and envoy configuration is abstracted away.
+
+```yaml
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: coraza-ingressgateway
+  namespace: istio-ingress
+spec:
+  imagePullPolicy: IfNotPresent
+  phase: AUTHN
+  pluginConfig:
+    default_directives: default
+    directives_map:
+      default:
+      - Include @demo-conf
+      - SecDebugLogLevel 9
+      - SecRuleEngine On
+      - Include @crs-setup-demo-conf
+      - Include @owasp_crs/REQUEST-941-APPLICATION-ATTACK-XSS.conf
```

Comment:
I think we are using the two examples to demonstrate the two usages, one rule vs entire rule set. If there isn't any particular reason for the current order, I would make this example load the entire rule set and the other one only one rule. I think the first example in a doc will be the one copy-pasted the most

Reply:
Good call! I only wanted to showcase that if you decide to include the wasm module at the ingress-gateway level and at the namespace level, that you could use less rules at the gateway, as it will affect all traffic and traffic which is blocked never makes it to your namespace.
Activating the entire crs at ingress-gateway and namespace might still make sense, as you also filter communication from namespace to namespace within the mesh.

Reply:
I changed for both using the entire rule set.
The sentence that already blocked traffic will not reach the namespace, should be enough in my opinion to clarify the use cases.